src/js/view/controls/components/timeslider.js

```diff
@@ -280,7 +280,6 @@ class TimeSlider extends Slider {
     }
 
     addCues(model, cues) {
-        this.resetChapters();
```

Comment:
I don't think this is safe. 
`api.setCues` changes the `cues` in `model`, which then calls this function. This function used to clear the already added cues. Removing this code will change the api behavior.

Reply:
@robwalch - Your thoughts on this? @pajong and I were talking. He's worried that is a customer calls `setCues()` which isn't documented, it would cause duplicate cues. But, we have to make sure that multiple parts of the codebase are able to call `setCues()` (i.e. ads and chapters).

Reply:
New thought: could change `setCues()` in controller.js to clear the cues if there are any before setting them. I believe the only thing that would be able to call `setCues()` would be the JavaScript API. The mixins and ad plugins themselves don't call it.

Reply:
The ads plugins call `setCues()`
e.g.
https://github.com/jwplayer/jwplayer-ads-vast/blob/master/src/js/vastPlugin.js#L468
https://github.com/jwplayer/jwplayer-ads-googima/blob/master/src/PlaylistItemManager.js#L181

Reply:
Yeah, I already realized that it wouldn't work. :(

Reply:
@jnatalzia - Curious your thoughts on this too.

Reply:
I don't have too much context on exactly what `resetChapters` here means but what's the risk around leaving it here to preserve existing functionality and adding it to `reset`?

Reply:
@jnatalzia - It would cause the issue that is the cause of this work which is that the chapter cues will be cleared when the ad plugins call `addCues()`.

Reply:
@waxidiotic I am concerned that this only works if setCues is only called twice when you have chapter cues and adds. Now you can't use `setCues` to reset the cues (API change which Jong pointed out.)

What we should do is add a `getCues` method if we don't have one, and the ads or chapters should get cues, add their cues to a copy of that array, and then set cues to the new updated list.